Support for custom store class

marden opened the pull request:
At the moment there is no way to specify custom store for Factory Muffin. 

This PR adds support for `storeClass` config variable to specify a class to be used as a store.

Module config example:
```
modules:
  config:
    DataFactory:
      factories: tests/_support/factories
      storeClass: \common\tests\store\MyCustomStore
```

Commit:
Support for custom store class

DavertMik commented:
Thanks @marden !

1. What if the store requires some arguments on creation? This config doesn't cover this case. Could you mention explicitly in the documentation that the class will be created via constructor with no extra params. Or maybe instead of a class name ask for a factory method so it could create any kind of store.
2. Could you provide usage example in documentation? Documentation is inside the same file in docblocks.

Commit:
Ability to use factory create method. Updated docs.

marden commented:
@DavertMik Thanks for the feedback. 

- Renamed parameter to `customStore`
- It can be a simple class or a factory with `create` method
- Updated doc.

marden commented:
@DavertMik Please review.

marden commented:
@DavertMik Hello?

DavertMik commented:
Thanks! Looks good to me!
Merging
